packages/@ember/object/type-tests/compat/dependent-key-compat.test.ts

```diff
@@ -0,0 +1,17 @@
+import { dependentKeyCompat } from '@ember/object/compat';
+import { expectTypeOf } from 'expect-type';
+
+expectTypeOf(dependentKeyCompat).toMatchTypeOf<PropertyDecorator>();
```

Comment:
Huh: why `toMatchTypeOf` rather than `toEqualTypeOf` here?

Reply:
Because it can also return a Decorator.